### ISSUE TITLE
[in_app_purchase] buyConsumable without internet connection causes screen freeze

### ISSUE THREAD

**Issue by nadrolinux** [flutter/flutter#136430]
### Is there an existing issue for this?

- [X] I have searched the [existing package issues](https://github.com/flutter/flutter/issues?q=is%3Aopen+is%3Aissue+label%3Apackage)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### What package does this bug report belong to?

in_app_purchase

### What target platforms are you seeing this bug on?

Android

### Have you already upgraded your packages?

Yes

### Steps to reproduce

1. Execute InAppPurchase.instance.buyConsumable or InAppPurchase.instance.buyNonConsumable with internet connection off.
2. Close Google Billing dialog contained 'you are offline' info by a system back button, outside dialog click or 'ok' button.
3. Google Billing dialog dissappear, but screen is like freezed. Even system back button doesn't work. It looks like a some fully transparent fullscreen widget still cover app widget (I didn't add any cover widget myself).

### Expected results

App widget is responsive, like before calling InAppPurchase.instance.buyConsumable or InAppPurchase.instance.buyNonConsumable.

### Actual results

I use Pixel 6a with Android 14. With internet connection turned on and when a purchase is finalized (payment both success and failed) all works fine. Screen freeze appears only when internet connection is off. Maybe InAppPurchase.instance.completePurchase do something related to blocking a screen, thats why a purchase flow works fine in a typical scenerio?

BTW. buyConsumable/buyNonConsumable always return true, even with internet connection off, so it looks like this bool works like a pure void. It would be nice to return false when internet connection is off or just report this error simply by InAppPurchase.instance.purchaseStream.listen. At now we don't get any error status of Flutter side when internet connection is off.

### Code sample

Just execute a buyConsumable/buyNonConsumable command with internet off and close an error dialog.

### Flutter Doctor output

[✓] Flutter (Channel stable, 3.13.4, on Fedora Linux 38 (Workstation Edition) 6.5.6-200.fc38.x86_64, locale pl_PL.UTF-8)
    • Flutter version 3.13.4 on channel stable at /data/sdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 367f9ea16b (4 tygodnie temu), 2023-09-12 23:27:53 -0500
    • Engine revision 9064459a8b
    • Dart version 3.1.2
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /data/sdk/android
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /data/sdk/android
    • Java binary at: /data/aplikacje/Android Studio/jbr/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /var/lib/flatpak/exports/bin/com.google.Chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 16.0.6 (Fedora 16.0.6-3.fc38)
    • cmake version 3.27.6
    • ninja version 1.11.1
    • pkg-config version 1.8.0

[✓] Android Studio (version 2022.3)
    • Android Studio at /data/aplikacje/Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] VS Code (version 1.83.0)
    • VS Code at /usr/share/code
    • Flutter extension version 3.74.0

[✓] Connected device (3 available)
    • Pixel 6a (mobile) • 23191JEGR02256 • android-arm64  • Android 14 (API 34)
    • Linux (desktop)   • linux          • linux-x64      • Fedora Linux 38 (Workstation Edition) 6.5.6-200.fc38.x86_64
    • Chrome (web)      • chrome         • web-javascript • Google Chrome 117.0.5938.149 unknown

[✓] Network resources
    • All expected network resources are available.

• No issues found!

**Comment by huycozy** [comment]
Hi @nadrolinux 
I can't reproduce this on Android 14 and also on Android 11 devices. Could you check if there is any crash or noticeable log from the output console? 
It would appreciated if you can share a recorded video as well. Thanks!

**Comment by nadrolinux** [comment]
Hi @huycozy 
I use latest version of IAP plugin -> 3.1.11. Situation is really interesting, because when I downloaded clean example all works fine. However when I copied all files from "working example" to my main project folder issue appears again (folder was empty, I removed all files, hidden too). When I changed my main project folder name from eg. xyz to xyz1 issue gone... It looks like some cache issue with VS Code/Flutter. Probably it's not a bug in a IAP library, but I will provide answer here when I'll find a solution for this crazy issue. Give me little more time.
